File: library-db/src/main/java/com/my/library_db/DBModuleInit.java
package com.my.library_db;

import android.app.Application;
import android.content.Context;

import com.my.library_base.init.IModuleInit;
import com.my.library_base.logs.GLog;
import com.my.library_db.db.DatabaseInit;


/**
 * Created by yan
 * 基础库自身初始化操作
 */

public class DBModuleInit implements IModuleInit {
    public static Context applicationContext;

    @Override
    public boolean onInitAhead(Application application) {
        GLog.i("DB层初始化 -- onInitAhead");
        initData(application);
        DatabaseInit.getInstance().init(application);
        return true;
    }

    @Override
    public boolean onInitLow(Application application) {
        GLog.i("DB层初始化 -- onInitLow");
        return true;
    }

    private void initData(Application application) {
        applicationContext = application.getApplicationContext();
    }

}

File: library-db/src/main/java/com/my/library_db/DBModuleInit.kt
package com.my.library_db

import android.app.Application
import android.content.Context
import com.my.library_base.init.IModuleInit
import com.my.library_base.logs.GLog
import com.my.library_db.db.DatabaseInit

/**
 * Created by yan
 * 基础库自身初始化操作
 */
class DBModuleInit : IModuleInit {
    override fun onInit(application: Application): Boolean {
        GLog.i("DB层初始化 -- onInitAhead")
        initData(application)
        DatabaseInit.instance!!.init(application)
        return true
    }

    override fun onDestroy(application: Application): Boolean {
        GLog.i("DB层销毁 -- onDestroy")
        return true
    }

    private fun initData(application: Application) {
        applicationContext = application.applicationContext
    }

    companion object {
        var applicationContext: Context? = null
    }
}